File: testdata/fixtures/ruby/CWE-532.rb
# CWE-532: Insertion of Sensitive Information into Log File
#
# Logging a password, secret or token exposes it in log files which are often
# world-readable or shipped to third parties.

def leak(password, secret, api_key)
  # ruleid: ruby-sensitive-log
  logger.info(password)
  # ruleid: ruby-sensitive-log
  puts secret
  # ruleid: ruby-sensitive-log
  Rails.logger.info(api_key)
end

def safe
  # ok: ruby-sensitive-log
  logger.info("user logged in")
  # ok: ruby-sensitive-log
  puts "request processed"
  # ok: ruby-sensitive-log
  Rails.logger.info("done")
end

File: testdata/fixtures/ocaml/CWE-532.ml
(* CWE-532: Insertion of Sensitive Information into Log File
 * Logging a password or secret exposes it in log files which are often
 * world-readable or shipped to third parties.
 *)

let vuln password secret api_key =
  (* ruleid: ocaml-sensitive-log *)
  let _ = print_endline password in
  (* ruleid: ocaml-sensitive-log *)
  let _ = print_string secret in
  (* ruleid: ocaml-sensitive-log *)
  let _ = Logs.info (fun m -> m "%s" api_key) in
  ()

let safe () =
  (* ok: ocaml-sensitive-log *)
  let _ = print_endline "user logged in" in
  (* ok: ocaml-sensitive-log *)
  let _ = print_string "request processed" in
  (* ok: ocaml-sensitive-log *)
  let _ = Logs.info (fun m -> m "done") in
  ()
